src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -31210,4 +31362,1076 @@ static Expression<Func<Func<string, string>>> Test()
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "M").WithArguments("E.extension(int).M()").WithLocation(5, 33),
+            // (6,32): error CS9302: 'E.extension(int).P': new protected member declared in an extension block
+            //         protected internal int P { get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "P").WithArguments("E.extension(int).P").WithLocation(6, 32),
+            // (7,44): error CS9302: 'E.extension(int).P2.get': new protected member declared in an extension block
+            //         public int P2 { protected internal get => 0; set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "get").WithArguments("E.extension(int).P2.get").WithLocation(7, 44),
+            // (8,54): error CS9302: 'E.extension(int).P3.set': new protected member declared in an extension block
+            //         public int P3 { get => 0; protected internal set { } }
+            Diagnostic(ErrorCode.ERR_ProtectedInExtension, "set").WithArguments("E.extension(int).P3.set").WithLocation(8, 54));
+    }
+
+    [Fact]
+    public void Validation_EntryPoint_01()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        public static void Main() { System.Console.Write("ran"); }
+    }
+}
+""";
+        var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+        comp.VerifyEmitDiagnostics();
+        // Tracked by https://github.com/dotnet/roslyn/issues/76130
+        // FindEntryPoint/NameSymbolSearcher skip "E" since it has not declaration named "Main"
+        // CompileAndVerify(comp, expectedOutput: "ran");
+    }
+
+    [Fact]
+    public void Validation_EntryPoint_02()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        public static void Main() { System.Console.Write("ran"); }
+    }
+}
+""";
+        var comp = CreateCompilation(source, options: TestOptions.ReleaseExe.WithMainTypeName("E"));
+        comp.VerifyEmitDiagnostics();
+        CompileAndVerify(comp, expectedOutput: "ran").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void Validation_EntryPoint_03()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        public static void Main() { System.Console.Write("ran"); }
+    }
+}
+""";
+        var comp = CreateCompilation(source, options: TestOptions.ReleaseExe.WithMainTypeName("E.<>E__0"));
+        comp.VerifyEmitDiagnostics(
+            // error CS1555: Could not find 'E.<>E__0' specified for Main method
+            Diagnostic(ErrorCode.ERR_MainClassNotFound).WithArguments("E.<>E__0").WithLocation(1, 1));
+    }
+
+    [Fact]
+    public void Validation_EntryPoint_04()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        public static void Main() { }
+    }
+}
+""";
+        var comp = CreateCompilation(source, options: TestOptions.ReleaseExe.WithMainTypeName("E."));
```

Comment:
>E. [](http://example.com/codeflow?start=95&length=2)

Did you mean to use just "E"?

Reply:
No, this is intentional. The current logic in `FindEntryPoint` splits this string and calls `GetNamespaceOrTypeByQualifiedName` with the parts. But that calls `GetMembers` which finds a member named "" (blank) which is the extension declaration.

There is a follow-up note above for behavior of entry points inside extension blocks (I think we'll likely want to disallow), and this test results in an error anyways, so I'm not planning to address/change the behavior for this test in this PR, but I wanted to point it out.
I could leave a follow-up note here too if you'd like.

Reply:
> I could leave a follow-up note here too if you'd like.

Consider adding a PROTOTYPE comment, I do not think ```FindEntryPoint``` should attempt to find types with empty name